### ISSUE TITLE
Support installation of Carvel-type Supervisor Service packages

### ISSUE THREAD

**Issue by rosenhouse** [vmware/govmomi#3624]
**Is your feature request related to a problem? Please describe.**
I want to use `govc` to install and activate a Carvel supervisor service.

**Describe the solution you'd like**
Enhance `govc namespace.service.create` and supporting library functions to work with Carvel manifests.

**Describe alternatives you've considered**
🤷 Writing my own Go library to drive the relevant APIs?  Using the Python SDK instead?

**Additional context**
It appears that #2668 and #2690 introduced partial support for `vsphere` type Supervisor Services, but not Carvel-type.

**Comment by dougm** [comment]
Looks like we need to add `carvel_spec`, currently only `vsphere_spec` is supported in govmomi: https://developer.broadcom.com/xapis/vsphere-automation-api/latest/vcenter/api/vcenter/namespace-management/supervisor-services/post/

**Comment by rosenhouse** [comment]
I'm interested in contributing this feature -- along with the ability to delete it, which implies that I'll need to be able to list, deactivate and delete the "versions" of the Supervisor Service since you can't delete the service without first deleting its versions: https://developer.broadcom.com/xapis/vsphere-automation-api/latest/vcenter/namespace_management/supervisor_services-versions/

What sort of test coverage would be appropriate for this set of features?

I'm expecting to develop against a real VC, using a script that runs `govc` to do some user workflows (e.g. install 2 versions, list the versions, remove them, and remove the service).  

I see some bats tests that look kinda like that, e.g. [here](https://github.com/vmware/govmomi/blob/main/govc/test/namespace.bats) but it looks like they depend on a test harness here [`vapi/namespace/simulator`](https://github.com/vmware/govmomi/blob/main/vapi/namespace/simulator/simulator.go) as well?  Am I understanding right?

**Comment by dougm** [comment]
Right, the bats tests run against [vcsim](https://github.com/vmware/govmomi/tree/main/vcsim), which includes the namespace simulator.  Clients (such as govc) talk to vcsim just as they would a real vCenter. Starting with a script against real vCenter is good, esp. if you can share it. Then myself and others could help with the simulator part if you need it.
